transport/control.go

```diff
@@ -189,6 +214,14 @@ func (f *inFlow) onRead(n uint32) uint32 {
 		return 0
 	}
 	f.pendingData -= n
+	if f.delta > 0 {
+		f.delta -= int32(n)
```

Comment:
I think this would be easier to read without the casts and taking the negative of a negative number:

```go
if n > f.delta {
  f.delta = 0
  n -= f.delta
} else {
  f.delta -= n
  n = 0
}
```

(Also would allow f.delta to be uint32 like it seems like it wants to be.)